Fix: 27 TypeScript typecheck errors in frontend

## Problem

Running `npm run check` in the frontend reports 27 TypeScript errors across 8 files.

## Affected Files

### 1. `src/components/Layout.svelte` (2 errors)
- `'mediaQuery' is possibly 'null'` at line 15
- `Property 'addListener' does not exist on type 'never'` at line 23

### 2. `src/components/__tests__/Layout.test.ts` (2 errors)
- `Cannot assign to 'matches' because it is a read-only property`
- `Conversion of type 'MediaQueryList' to type 'MediaQueryListEvent' may be a mistake`

### 3. `src/components/__tests__/Nav.test.ts` (1 error)
- `'call' is possibly 'null'`

### 4. `src/components/search/__tests__/SearchBar.test.ts` (6 errors)
- `Cannot find module 'module'`
- Multiple `Mock` type assignment errors
- `Spread types may only be created from object types`

### 5. `src/components/search/__tests__/SearchResults.test.ts` (1 error)
- `Cannot find module 'module'`

### 6. `src/stores/__tests__/pwaStore.test.ts` (7 errors)
- Multiple `Cannot find name 'global'` errors
- `Cannot find name 'Buffer'`

### 7. `src/stores/__tests__/websocketStore.test.ts` (1 error)
- `Cannot find module 'module'`

### 8. `src/test/setup.ts` (7 errors)
- Implicit `any` types for event handler parameters
- `Cannot assign to 'matches' because it is a read-only property`
- `MediaQueryList` to `MediaQueryListEvent` type conversion issues

## Suggested Fixes

1. **Add `@types/node`** to devDependencies (fixes `global`, `Buffer`, `module` errors)
2. **Add null checks** for `mediaQuery` in Layout.svelte
3. **Fix MediaQueryList mocking** - use proper type casting or create a mutable mock object
4. **Add explicit types** to event handler parameters in test/setup.ts
5. **Fix Mock type annotations** in SearchBar.test.ts

--- a/frontend/package.json
+++ b/frontend/package.json
@@ -18,6 +18,7 @@
     "@sveltejs/vite-plugin-svelte": "^3.0.3",
     "@testing-library/jest-dom": "^6.9.1",
     "@testing-library/svelte": "^5.3.1",
+    "@types/node": "^25.0.9",
     "@typescript-eslint/eslint-plugin": "^6.21.0",
     "@typescript-eslint/parser": "^6.21.0",
     "eslint": "^8.56.0",

--- a/frontend/src/components/Layout.svelte
+++ b/frontend/src/components/Layout.svelte
@@ -12,7 +12,7 @@
     uiStore.setIsMobile(mediaQuery.matches);
 
     const handleResize = (event?: MediaQueryListEvent) => {
-      uiStore.setIsMobile(event?.matches ?? mediaQuery.matches);
+      uiStore.setIsMobile(event?.matches ?? mediaQuery?.matches ?? false);
     };
 
     handleResize();
@@ -20,8 +20,9 @@
       mediaQuery.addEventListener('change', handleResize);
       cleanupListener = () => mediaQuery?.removeEventListener('change', handleResize);
     } else {
-      mediaQuery.addListener(handleResize);
-      cleanupListener = () => mediaQuery?.removeListener(handleResize);
+      // Legacy browsers support addListener/removeListener
+      (mediaQuery as MediaQueryList).addListener(handleResize);
+      cleanupListener = () => (mediaQuery as MediaQueryList | null)?.removeListener(handleResize);
     }
   }
 

--- a/frontend/src/components/__tests__/Layout.test.ts
+++ b/frontend/src/components/__tests__/Layout.test.ts
@@ -8,15 +8,15 @@ const { default: Layout } = await import('../Layout.svelte');
 describe('Layout', () => {
   it('sets isMobile based on matchMedia changes', async () => {
     const listeners: Array<(event: MediaQueryListEvent) => void> = [];
-    const media = {
+    const media: { matches: boolean } & Omit<MediaQueryList, 'matches'> = {
       matches: true,
       media: '(max-width: 1023px)',
       onchange: null,
-      addEventListener: (_event: string, handler: (event: MediaQueryListEvent) => void) => {
-        listeners.push(handler);
+      addEventListener: (_event: string, handler: EventListenerOrEventListenerObject | null) => {
+        listeners.push(handler as (event: MediaQueryListEvent) => void);
       },
-      removeEventListener: (_event: string, handler: (event: MediaQueryListEvent) => void) => {
-        const index = listeners.indexOf(handler);
+      removeEventListener: (_event: string, handler: EventListenerOrEventListenerObject | null) => {
+        const index = listeners.indexOf(handler as (event: MediaQueryListEvent) => void);
         if (index >= 0) {
           listeners.splice(index, 1);
         }
@@ -24,8 +24,8 @@ describe('Layout', () => {
       addListener: () => {},
       removeListener: () => {},
       dispatchEvent: () => true,
-    } as MediaQueryList;
-    const matchMediaSpy = vi.spyOn(window, 'matchMedia').mockReturnValue(media);
+    };
+    const matchMediaSpy = vi.spyOn(window, 'matchMedia').mockReturnValue(media as MediaQueryList);
     let isMobile = false;
     const unsubscribe = uiStore.subscribe((state) => {
       isMobile = state.isMobile;
@@ -37,7 +37,7 @@ describe('Layout', () => {
     expect(isMobile).toBe(true);
 
     media.matches = false;
-    listeners.forEach((handler) => handler(media as MediaQueryListEvent));
+    listeners.forEach((handler) => handler({ matches: false } as MediaQueryListEvent));
     expect(isMobile).toBe(false);
     matchMediaSpy.mockRestore();
     unsubscribe();

--- a/frontend/src/components/__tests__/Nav.test.ts
+++ b/frontend/src/components/__tests__/Nav.test.ts
@@ -20,7 +20,7 @@ describe('Nav', () => {
     await fireEvent.click(button);
 
     expect(setActiveSpy).toHaveBeenCalled();
-    const call = setActiveSpy.mock.calls[0][0];
-    expect(call.id).toBe('section-2');
+    const call = setActiveSpy.mock.calls[0]?.[0];
+    expect(call?.id).toBe('section-2');
   });
 });

--- a/frontend/src/components/search/__tests__/SearchBar.test.ts
+++ b/frontend/src/components/search/__tests__/SearchBar.test.ts
@@ -1,28 +1,34 @@
-import { describe, it, expect, vi, beforeEach, afterEach } from 'vitest';
+import { describe, it, expect, vi, beforeEach, afterEach, type Mock } from 'vitest';
 import { render, fireEvent, screen, cleanup } from '@testing-library/svelte';
 import { createRequire } from 'module';
+import type { Writable } from 'svelte/store';
 
 const require = createRequire(import.meta.url);
 const { writable } = require('svelte/store') as typeof import('svelte/store');
 
+interface SearchState {
+  query: string;
+  scope: string;
+}
+
 const storeRefs: {
-  activeSection: ReturnType<typeof writable>;
-  state: ReturnType<typeof writable>;
+  activeSection: Writable<{ id: string; name: string } | null>;
+  state: Writable<SearchState>;
   searchStore: {
-    subscribe: ReturnType<typeof writable>['subscribe'];
-    setQuery: ReturnType<typeof vi.fn>;
-    setScope: ReturnType<typeof vi.fn>;
-    search: ReturnType<typeof vi.fn>;
-    clear: ReturnType<typeof vi.fn>;
+    subscribe: Writable<SearchState>['subscribe'];
+    setQuery: Mock;
+    setScope: Mock;
+    search: Mock;
+    clear: Mock;
   };
-} = {} as any;
+} = {} as typeof storeRefs;
 
 vi.mock('../../../stores', () => {
   storeRefs.activeSection = writable<{ id: string; name: string } | null>({
     id: 'section-1',
     name: 'Music',
   });
-  storeRefs.state = writable({
+  storeRefs.state = writable<SearchState>({
     query: '',
     scope: 'section',
   });

--- a/frontend/src/test/setup.ts
+++ b/frontend/src/test/setup.ts
@@ -4,14 +4,14 @@ import { vi } from 'vitest';
 if (!window.matchMedia) {
   window.matchMedia = vi.fn().mockImplementation((query: string) => {
     let listeners: Array<(event: MediaQueryListEvent) => void> = [];
-    const mediaQueryList: MediaQueryList = {
+    const mediaQueryList: { matches: boolean } & Omit<MediaQueryList, 'matches'> = {
       matches: false,
       media: query,
       onchange: null,
-      addEventListener: (_event, handler) => {
+      addEventListener: (_event: string, handler: EventListenerOrEventListenerObject | null) => {
         listeners.push(handler as (event: MediaQueryListEvent) => void);
       },
-      removeEventListener: (_event, handler) => {
+      removeEventListener: (_event: string, handler: EventListenerOrEventListenerObject | null) => {
         listeners = listeners.filter((item) => item !== handler);
       },
       addListener: () => {},
@@ -20,9 +20,9 @@ if (!window.matchMedia) {
     };
     (mediaQueryList as unknown as { trigger: (matches: boolean) => void }).trigger = (matches: boolean) => {
       mediaQueryList.matches = matches;
-      listeners.forEach((handler) => handler(mediaQueryList as MediaQueryListEvent));
+      listeners.forEach((handler) => handler({ matches } as MediaQueryListEvent));
     };
-    return mediaQueryList;
+    return mediaQueryList as MediaQueryList;
   }) as unknown as typeof window.matchMedia;
 }
 

